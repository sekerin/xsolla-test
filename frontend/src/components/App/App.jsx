import React, { Component }    from 'react';
import PropTypes               from 'prop-types';
import { Link } from 'react-router-dom';

import Grid   from 'react-bootstrap/lib/Grid';
import Navbar from 'react-bootstrap/lib/Navbar';

const propTypes = {
  children: PropTypes.node
};

class App extends Component {
  render() {
    return (
      <div>
        <Navbar>
          <Navbar.Header>
            <Navbar.Brand>
              <Link to='/'>File Manager</Link>
            </Navbar.Brand>
          </Navbar.Header>
        </Navbar>
        <Grid>
          Hello, world!
        </Grid>
      </div>
    );
  }
}

App.propTypes = propTypes;

export default App;
