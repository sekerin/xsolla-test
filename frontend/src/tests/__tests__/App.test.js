import React            from 'react';
import { StaticRouter } from 'react-router-dom';

import { mount }       from 'enzyme';
import { mountToJson } from 'enzyme-to-json';

import App from '../../components/App/index';

it('renders without crashing', () => {
  const output = mount(<StaticRouter context={{}}><App/></StaticRouter>);
  expect(mountToJson(output)).toMatchSnapshot();
});
