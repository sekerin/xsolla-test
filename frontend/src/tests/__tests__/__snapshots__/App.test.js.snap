// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App>
      <div>
        <Uncontrolled(Navbar)
          bsStyle="default"
        >
          <Navbar
            bsClass="navbar"
            bsStyle="default"
            collapseOnSelect={false}
            componentClass="nav"
            fixedBottom={false}
            fixedTop={false}
            fluid={false}
            inverse={false}
            onToggle={[Function]}
            staticTop={false}
          >
            <nav
              className="navbar navbar-default"
            >
              <Grid
                bsClass="container"
                componentClass="div"
                fluid={false}
              >
                <div
                  className="container"
                >
                  <NavbarHeader>
                    <div
                      className="navbar-header"
                    >
                      <NavbarBrand>
                        <Link
                          className="navbar-brand"
                          replace={false}
                          to="/"
                        >
                          <a
                            className="navbar-brand"
                            href="/"
                            onClick={[Function]}
                          >
                            File Manager
                          </a>
                        </Link>
                      </NavbarBrand>
                    </div>
                  </NavbarHeader>
                </div>
              </Grid>
            </nav>
          </Navbar>
        </Uncontrolled(Navbar)>
        <Grid
          bsClass="container"
          componentClass="div"
          fluid={false}
        >
          <div
            className="container"
          >
            Hello, world!
          </div>
        </Grid>
      </div>
    </App>
  </Router>
</StaticRouter>
`;
